pyproject.toml

```diff
@@ -15,8 +15,7 @@ expand = {"pex_version" = "version"}
 
 [project]
 name = "pex"
-dynamic = ["version"]
-requires-python = ">=2.7,<3.13,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*"
+dynamic = ["version", "requires-python"]
```

Comment:
If this is now required I'm not a fan. The intent is to record the real requires-python here in the static metadata and only override in CI tests. Moving the real metadata to the metadata_hook obscures a bit too much for my liking.

Reply:
The issue is that the spec says that tools must adhere to what is defined statically (e.g. they cannot be changed) and Hatchling had a bug that it was not enforcing that.

If you would prefer I can add an option to enable that unsafe behavior but it would be bad for the ecosystem. What do you think?

FYI here is a comment that explains a little bit https://discuss.python.org/t/respecting-core-metadata-2-2-when-building-from-source-distributions/48886/60

It's not exactly related but the bug was revealed during that discussion/thread.

Reply:
Ok, I think I grok the thread you pointed me to. IIUC the criux lies in PEP-621 under https://peps.python.org/pep-0621/#dynamic:

> * Build back-ends MUST raise an error if the metadata specifies a field statically as well as being listed in dynamic.

If that were, instead:
```
* When a field is specified statically as well as marked dynamic, its value must be obtained from the backend dynamic generation process.
```

All would be well.

@ofek I'll send up a PR and add you as a reviewer that sortof acheives the end I want here using non-canonical fields in pyproject.toml to carry the static metadata / template metadata. That PR may contain comments about the PyPA that you want no association with; so feel free to not comment if you're uncomfortable!

Thanks for hunting down users affected by your rocky release period though, I really appreciate it.